ui/src/components/Equipment/PlateManipulatorMaintenance.js

```diff
@@ -0,0 +1,210 @@
+              type={this.props.inputType}
+              onChange= {(e) => { this.handleInputChange(e); }}
+              ref={(ref) => { this.inputRef = ref; }}
+            />
+            {React.createElement(PlateManipulatorActionButton,
+              {
+                label: this.props.btn_label,
+                cmd: this.props.cmd,
+                args: this.state.input_value,
+                enabled: true,
+                send_command: this.props.send_command,
+                key: this.props.btn_label
+            })}
+          </InputGroup>
+        </Form.Group>
+      </Form>
+    </span>);
+
+    input = React.cloneElement(input, props);
+
+    return input;
+  }
+
+  render() {
+    return (
+      React.createElement(PlateManipulatorActionGroup,
+        {
+          name: [`${this.props.header_msg} : ${this.props.value}`, this.props.viewComponent],
+          actionComponent: this.actionComponent(),
+          key: `${this.props.header_msg}:${this.props.value}`
+        })
+    );
+  }
+}
+
+export default class PlateManipulatorMaintenance extends React.Component {
+  buildActionButton(cmdinfo) {
+    return React.createElement(PlateManipulatorActionButton,
+      {
+        label: cmdinfo[1],
+        cmd: cmdinfo[0],
+        args: cmdinfo[3],
+        enabled: this.props.commands_state[cmdinfo[0]],
+        send_command: this.props.send_command,
+        key: cmdinfo[1]
+      });
+  }
+
+  buildActionGroup(grpinfo) {
+    const butgrp = [];
+
+    for (const cmdinfo of grpinfo[1]) {
+      butgrp.push(this.buildActionButton(cmdinfo));
+    }
+
+    return React.createElement(PlateManipulatorActionGroup,
+      {
+        name: grpinfo[0],
+        actionComponent: <ButtonGroup>{butgrp}</ButtonGroup>,
+        key: grpinfo[0]
+      });
+  }
+
+  render() {
+    const groups = [];
+    let msg = '';
+
+    if (Object.keys(this.props.commands).length !== 0
+        && this.props.commands.cmds !== 'SC maintenance controller not defined') {
+      for (const cmdgrp of this.props.commands.cmds) {
+        groups.push(this.buildActionGroup(cmdgrp));
+      }
+    } else {
+      return (
+        <div />
+      );
+    }
+
+
+    if (this.props.message !== '') {
+      msg = this.props.message;
+    }
+
+    let scan_limits = '';
+    if (this.props.global_state.scan_limits) {
+      scan_limits = `[ ${this.props.global_state.scan_limits.toString().split(',').join(',   ')}]`;
+    }
+
+    let plateBarcode = '';
+    if (this.props.global_state.plate_info) {
+      plateBarcode = this.props.global_state.plate_info.plate_barcode.toString();
+    }
+
+    return (
+      <div>
+        { groups }
+        { msg ? (
+          <Card className='mb-2'>
+            <Card.Header>Status message</Card.Header>
+            <Card.Body>
+              <span className="scMessage">{ msg }</span>
+            </Card.Body>
+          </Card>
+        ) : null
+        }
+        {React.createElement(PlateManipulatorAction,
```

Comment:
Why do you need to use createElement ?

Reply:
Normally i don't use createElement, This PR is new but this class was created 4 years ago and it has similar function to SampleChanger.js and SampleChangerMaintenance.js , that both use createElement.
 is there  reports of createElement being deprecated? maybe we should change it the 4 files and create a new PR "removing createElement" .